### PR TITLE
MM-17912 - slash command to exercise FileInfo search API

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

- Demonstrate usage of the `GetWithOptions` method to get FileInfos that was added in https://github.com/mattermost/mattermost-server/pull/13647/ 

#### Ticket Link

- Part of https://github.com/mattermost/mattermost-server/issues/12699 and
https://mattermost.atlassian.net/browse/MM-17912

### CODE REVIEW

**Comment by ashishbhate** [comment]
CI will fail because https://github.com/mattermost/mattermost-server/pull/13647/  hasn't been merged yet. 

On my local machine I have updated the `go.mod` file to point to my fork of the mattermost-server that has the required changes. You can do this by running
`go mod edit -replace=github.com/mattermost/mattermost-server/v5=github.com/ashishbhate/mattermost-server/v5@25037fd7`

**Comment by hanzei** [comment]
@ashishbhate Yes, please update `go.mod` to make the build pass. :+1:

**Comment by ashishbhate** [comment]
@hanzei I've updated the `go.mod` file.

File: server/command_hooks.go [hunk 1]
```diff
@@ -509,3 +521,60 @@ func (p *Plugin) crash() {
...
+			p.API.LogError(errorMessage, "err", userErr.Error())
+			return &model.CommandResponse{
+				ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+				Text:         errorMessage,
+			}
+		}
+		attachments = append(attachments,
+			&model.SlackAttachment{
+				Title:     f.Name,
+				TitleLink: permaLink + f.PostId,
```
Comment (ali-farooq0) [inline server/command_hooks.go]: this is really cool! It'll be great to include this file's public url as well. As I imagine most people using this api would want to display it's public url, and looking at `FileInfo` struct it's not very clear how would one go about doing that, so it'll be great to include an example here.

Comment (ashishbhate) [reply server/command_hooks.go]: Do you mean in addition to the link to the post Id or instead of the link to the post id?

**Comment by ashishbhate** [comment]
I've updated the PR based on your feedback:

1. https://github.com/mattermost/mattermost-server/pull/13647 has been merged so I've updated the `go.mod` file to use upstream mattermost-server instead of my fork. There were some unrelated dependency updates as well. 

2. I've updated the slash command to show the file's public link. The server config will need to be set to allow public links: FileSettings>EnablePublicLink
I've made this a required configuration in this plugin's manifest

3. I've updated the README to include the added slash command

**Comment by jasonblais** [comment]
@ali-farooq0 Reminder to help with your review when you're back :)

File: server/command_hooks.go (hunk 1 above)
Comment (ali-farooq0) [reply server/command_hooks.go]: in addition works, looks like you already made the change, excellent! 👌

**Comment by ali-farooq0** [comment]
@prapti this is ready for testing

**Pull request merged**